src/woodbury.jl

```diff
@@ -76,6 +76,17 @@ function inv(W::AbstractWoodbury)
     return Woodbury(A′, U′, C′, V′)
 end
 
+# multiplying everything out would be O(m^2 k + m k^2) operations
+# best I could think of was O(m^2 k) operations
+function diag(W::AbstractWoodbury)
+    d = diag(W.A)
+    (k,m) = size(W.U)
+    for i = 1:k
+        d[i] += W.U[i,:]' * W.C * W.V[:,i]
```

Comment:
Would
```julia
d = diag(W.A) + vec(sum(W.U .* (W.C * W.V)', dims=2))
```
be faster?

Reply:
I tested this on some large matrices.  I've seen it go twice as fast empirically for large matrices though my math suggests that it should be the same.  It's also more concise.

This is my first pull request.  I pushed the change into my branch.  As far as I can tell, I don't need to do anything else but would be happy to if needed.